src/puppetlabs/puppetdb/pql/transform.clj

```diff
@@ -123,6 +123,10 @@
             (second arg)
             (vec (rest arg)))))])
 
+(defn transform-field
+  [& args]
+  (str/join #"." args))
```

Comment:
I'm surprised this works. It looks like it does because `str` is called on the regex. I think you probably just intended to have a string there.